Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/tabs/location/ContactDiaryLocationListViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package de.rki.coronawarnapp.contactdiary.ui.day.tabs.location
+
+import androidx.lifecycle.asLiveData
+import com.squareup.inject.assisted.Assisted
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.contactdiary.model.Location
+import de.rki.coronawarnapp.contactdiary.storage.ContactDiaryRepository
+import de.rki.coronawarnapp.contactdiary.util.SelectableItem
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import org.joda.time.Instant
+
+class ContactDiaryLocationListViewModel @AssistedInject constructor(
+    @Assisted selectedDay: Long,
+    contactDiaryRepository: ContactDiaryRepository
+) : CWAViewModel() {
+
+    private val dayElement = contactDiaryRepository.filterForDay(Instant.ofEpochMilli(selectedDay))
+    private val selectableLocations = MutableStateFlow<List<SelectableItem<Location>>>(emptyList())
```

Comment:
don't really like how the whole selectability is handled, maybe change this to directly integrate with the room store later?

Reply:
Yeah it could directly go through the database. A flow of all locations, and a flow of the locations on this day, then combine them?

Reply:
sounds good, will revisit when db is done 👍